### ISSUE TITLE
Failed backup not reported in action output

### ISSUE THREAD

**Issue by cjohnston1158** [canonical/postgresql-k8s-operator#242]
A backup can't be done on the leader, however if it is attempted, the action output does not note the failure:

```
ubuntu@node10:~$ juju run -m postgresql postgresql-k8s/leader list-backups
Running operation 5 with 1 task
  - task 6 on unit-postgresql-k8s-1

Waiting for task 6...
backups: |-
  backup-id             | backup-type  | backup-status
  ----------------------------------------------------

ubuntu@node10:~$ juju run -m postgresql postgresql-k8s/leader create-backup
Running operation 7 with 1 task
  - task 8 on unit-postgresql-k8s-1

Waiting for task 8...
ubuntu@node10:~$ juju run -m postgresql postgresql-k8s/leader list-backups
Running operation 9 with 1 task
  - task 10 on unit-postgresql-k8s-1

Waiting for task 10...
backups: |-
  backup-id             | backup-type  | backup-status
  ----------------------------------------------------
```

debug-log:

```
unit-postgresql-k8s-1: 15:11:56 ERROR unit.postgresql-k8s/1.juju-log Backup failed: Unit cannot perform backups as it is the cluster primary
```

**Comment by taurus-forever** [comment]
Dear @cjohnston1158 ,

Thank you for the bug report!. Do you have TLS in use?
https://charmhub.io/postgresql-k8s/docs/h-create-and-list-backups
```
Note: the action create-backup is limited to non-Primary units IF relation with TLS operator created. To avoid unnecessary service downtime always use non-Primary units for the action create-backup.
```

@marceloneppel can we improve UX here (to explain it better in error message and maybe set some temporary unit status "backup failed, see explanation in 'juju debug-log'...")? Tnx!

**Comment by cjohnston1158** [comment]
Yes, I have TLS in use, and I do understand that it can't be run on the primary, I would just expect feedback on an unsuccessful event.

**Comment by cjohnston1158** [comment]
Same with restore-backup if you haven't scaled to 1 unit.

```
$ juju run -m postgresql postgresql-k8s/leader restore backup-id=2023-09-16T23:50:47Z
Running operation 13 with 1 task
  - task 14 on unit-postgresql-k8s-1

Waiting for task 14...
```

`unit-postgresql-k8s-1: 23:56:47 ERROR unit.postgresql-k8s/1.juju-log Restore failed: Unit cannot restore backup as there are more than one unit in the cluster`

**Comment by dragomirp** [comment]
Hi, @cjohnston1158, we seem to be failing the events correctly. I've created a bug with juju for this https://bugs.launchpad.net/juju/+bug/2037279. Will update the ticket further when more details become available.

**Comment by jack-w-shaw** [comment]
Hi,

I'm unsure if this would count as a bug or a (well justified) feature request in Juju. We don't return explicitly by default whether a task succeeds or not in Juju, only printing the results from `event.set_result(res)`

This may be fixed/improved in a future patch release of 3.1. I will keep you posted here.

In the meantime, you may want to consider adding a line like `event.set_result({"msg": validationMessage})` around your `event.fail()` lines such as:
https://github.com/canonical/postgresql-k8s-operator/blob/main/src/backups.py#L397

**Comment by jack-w-shaw** [comment]
Hi again.

This has been discussed a little in the Juju team, and we are going to add fail messages into the default tabular output of `juju run`, hopefully by the next point release of 3.1

**Comment by taurus-forever** [comment]
The reported issue has been fixed on Juju side for 3.1.7+ https://bugs.launchpad.net/juju/+bug/2037279

Tested locally:
```
ubuntu@juju317:~$ juju run postgresql-k8s/0 create-backup
Running operation 20 with 1 task
  - task 21 on unit-postgresql-k8s-0

Waiting for task 21...
Action id 21 failed: Unit cannot perform backups as it is the cluster primary

ubuntu@juju317:~$ juju version
3.1.8-genericlinux-amd64
```

Also proper error messages shown for not-yet configured env:
```
ubuntu@juju317:~$ juju run postgresql-k8s/leader list-backups
Running operation 1 with 1 task
  - task 2 on unit-postgresql-k8s-0

Waiting for task 2...
Action id 2 failed: Relation with s3-integrator charm missing, cannot create/restore backup.

ubuntu@juju317:~$ juju run postgresql-k8s/leader create-backup
Running operation 3 with 1 task
  - task 4 on unit-postgresql-k8s-0

Waiting for task 4...
Action id 4 failed: Stanza was not initialised

```

Also we have improved official documentation to highlight limitations: https://charmhub.io/postgresql-k8s/docs/h-create-backup
```
Tip: To avoid unnecessary service downtime, always use non-primary units for the action create-backup. Keep in mind that:

TLS enabled: disables the command from running on primary units.
TLS not enabled: disables the command from running on non-primary units.
```

Resolved. Thank you @cjohnston1158 for reporting!
